private/PROPOSAL/Propagator.cxx

```diff
@@ -151,85 +152,117 @@ Propagator::Propagator(
     }
 
     nlohmann::json json_global;
-    if(json_config.contains("global")){
+    if (json_config.contains("global")) {
         json_global = json_config["global"];
 
         global_seed = json_global.value("seed", global_seed_);
-        do_interpolation = json_global.value("do_interpolation", do_interpolation_);
+        do_interpolation
+            = json_global.value("do_interpolation", do_interpolation_);
         uniform = json_global.value("uniform", uniform_);
 
-        sec_def_global.reset(new Sector::Definition(json_global));
+
+        if(json_global.contains("cuts")){
+            sec_def_global->cut_settings = EnergyCutSettings(json_global.at("cuts"));
+            sec_def_global->location = Sector::ParticleLocation::InsideDetector;
+        }
+
+        std::string scattering_model_str
+            = json_global.value("scattering", "highlandintegral");
+        sec_def_global->scattering_model
+            = ScatteringFactory::Get().GetEnumFromString(scattering_model_str);
+        sec_def_global->do_stochastic_loss_weighting
+            = json_global.value("stochastic_loss_weighting", false);
+        sec_def_global->stochastic_loss_weighting
```

Comment:
asking two times for "stochastic_loss_weighting" seems to be a copy-paste-mistake

Reply:
Done in e09cb5f